WordPress/gutenberg
# Patterns: Remove `reusable` text from menu once rename hint has been dismissed

## What?
Removes the `/reusable block` text from the `Create pattern` menu once the rename hint has been dismissed. Also sets the default sync status to `Synced` to better follow the existing reusable block workflow.

## Why?
Prevents the additional text hanging around for a full release cycle as the text is not used anywhere else in the interface it is only intended to be a temporary discovery mechanism. See #52194

## How?
Checks to see if the reusable block rename hint preference has been toggled off before displaying `Create pattern` menu.


## Testing Instructions

- Either use a new user account that hasn't toggle of the reusable block rename hint, or run the following script in the browser console to toggle the hint on `window.wp.data.dispatch(window.wp.preferences.store).set('core', 'isResuableBlocksrRenameHintVisible', true);`
- Add a paragraph block and in block overflow menu check that the Create pattern menu says `Create pattern/reusable block`
- Select menu to open the modal, and dismiss the rename hint notice
- Close the modal and open the block menu again and check that it now says `Create pattern`
- Reload the browser and open the Create pattern modal again and make sure that the default sync status is toggled on
- Save a few synced and unsynced blocks and make sure they behave as expected.
- Reset the hint again with `window.wp.data.dispatch(window.wp.preferences.store).set('core', 'isResuableBlocksrRenameHintVisible', true);` and then go to `/wp-admin/edit.php?post_type=wp_block` and add a new pattern and make sure the rename hint displays correctly in the model

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/3629020/8991e5c8-9835-4876-a462-018746d3b3bd
